src/main/java/com/griddynamics/cloud/learning/dal/repositories/ArticleRepository.java

```diff
@@ -0,0 +1,178 @@
+            Logger logger = Logger.getLogger(ArticleRepository.class.getName());
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+
+    }
+
+    @Override
+    public void update(Article item, String columnName, String newValue) {
+
+        String query = "UPDATE " + getEntityTableName() + " SET " + columnName + " = \'" + newValue +
+                "\' WHERE id = " + item.getId() + ";";
+
+        try (Connection connection = DBConnectionManager.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+
+            statement.executeUpdate();
+
+        } catch (SQLException e) {
+            Logger logger = Logger.getLogger(ArticleRepository.class.getName());
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+    }
+
+    public Optional<List<Article>> getFree() {
+
+        String query = "SELECT * FROM " + getEntityTableName() + " WHERE price = 0.0;";
+
+        return getArticles(query);
+    }
+
+    public Optional<List<Article>> getPaid() {
+
+        String query = "SELECT * FROM " + getEntityTableName() + " WHERE NOT price = 0.0;";
+
+        return getArticles(query);
+    }
+
+    public Optional<List<Article>> getArticlesByAuthor(long author_id) {
+
+        String query = "SELECT * FROM " + getEntityTableName() + " WHERE author_id = " + author_id + ";";
+
+        return getArticles(query);
+    }
+
+    public Optional<List<Article>> getArticlesByCurrency(String currency) {
+
+        String query = "SELECT * FROM " + getEntityTableName() + " WHERE currency = \'" + currency + "\';";
+
+        return getArticles(query);
+    }
+
+    public Optional<List<Article>> getBySpecifiedDate(LocalDateTime date) {
+
+        String query = "SELECT * FROM " + getEntityTableName() + " WHERE date = \'" + date.toLocalDate() + "\';";
+
+        return getArticles(query);
+    }
+
+    public Optional<List<Article>> getAfterSpecifiedDate(LocalDateTime date) {
+
+        String query = "SELECT * FROM " + getEntityTableName() + " WHERE date > \'" + date.toLocalDate() + "\';";
+
+        return getArticles(query);
+    }
+
+    public Optional<Article> findByTitle(String title) {
+
+        String query = "SELECT * FROM " + getEntityTableName() + " WHERE caption = \'" + title + "\';";
+
+        try (Connection connection = DBConnectionManager.getConnection();
+             PreparedStatement statement = connection.prepareStatement(query)) {
+
+            ResultSet resultSet = statement.executeQuery();
+            resultSet.next();
+            Article article = convertToEntity(resultSet);
+
+            return Optional.ofNullable(article);
+
+        } catch (SQLException e) {
+            Logger logger = Logger.getLogger(ArticleRepository.class.getName());
+            logger.log(Level.SEVERE, e.getMessage(), e);
+        }
+
+        return Optional.empty();
+    }
+
+    private Optional<List<Article>> getArticles(String query) {
```

Comment:
Like it 👍 
But we are able to to it even more reusable. What do use think about following signature 
private Optional<List<T>> getArticles(String query) 
It could be implemented in Repository class